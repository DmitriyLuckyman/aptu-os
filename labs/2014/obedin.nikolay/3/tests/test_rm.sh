#!/usr/bin/env bash

source $(dirname $0)/common.sh

test() {
    run ./import    $FS_DIR $FILES_DIR/file1 /file1
    run ./import    $FS_DIR $FILES_DIR/file2 /file2
    run ./mkdir     $FS_DIR /test
    run ./copy      $FS_DIR /file1 /test/file1
    run ./copy      $FS_DIR /file2 /test/file2

    run ./ls        $FS_DIR /
    run ./rm        $FS_DIR /file1
    run ./rm        $FS_DIR /test/file2
    run ./ls        $FS_DIR /
    run ./ls        $FS_DIR /test

    run ./mkdir     $FS_DIR /test/test2
    run ./copy      $FS_DIR /file2 /test/test2/file2
    run ./ls        $FS_DIR /test
    run ./ls        $FS_DIR /test/test2
    run ./rm        $FS_DIR /test/
    run ./ls        $FS_DIR /
    run ./rm        $FS_DIR /
    run ./ls        $FS_DIR /

    run ./rm        $FS_DIR /non_existent_dir
    run ./import    $FS_DIR $FILES_DIR/file2 /file1
    run ./copy      $FS_DIR /file1 /file2
    run ./copy      $FS_DIR /file1 /file3
    run ./copy      $FS_DIR /file1 /file4
}
