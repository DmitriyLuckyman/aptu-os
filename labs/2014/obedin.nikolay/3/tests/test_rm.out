D /
D test
F file1
F file2

D /
D test
F file2

D test
F file1

D test
D test2
F file1

D test2
F file2

D /
F file2

D /

Error: source file or dir not found

